### PR TITLE
feat: migrate `util/response` to new Submission module (service, utils, etc) 

### PR DESCRIPTION
**Pull request by karrui**
## Problem

This PR migrates `response` utility file to Typescript. Also refactored some typings and their references.

Related to #8.

## Solution

**Features**:

- Migrate `util/response` to new `SubmissionService` located in `module/submission`. This module should also be used when it is time to refactor the various submission related controllers.

**Improvements**:

- Use `BaseField` enum to instantiate Basic and MyInfo field constants in `shared/resources`

**Tests**:
- Add tests for `submission.service` and `submission.utils`

### CODE REVIEW

**Comment by karrui** [comment]
Tests are failing because the form document was not created properly (`validateBeforeSave` was turned off) :(

File: src/app/modules/submission/submission.utils.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    case ResponseMode.Encrypt:
+      return encryptModeFilter
+    default:
+      throw Error('getResponsesForEachField: Invalid response mode parameter')
+  }
+}
+
+const emailModeFilter = <T extends ModeFilterParam>(responses: T[]) => {
+  return responses.filter(
+    ({ fieldType }) => !FIELDS_TO_REJECT.includes(fieldType),
```
Comment (karrui) [inline src/app/modules/submission/submission.utils.ts:21]: I'm thinking of not using this import and just forcing a type like in `encryptModeFilter`, but I'm also afraid of splintering the sources of truth. Not a fan of this `FIELDS_TO_REJECT` constant though, KIV.

File: src/types/field/emailField.ts [hunk 2]
```diff
@@ -13,4 +13,6 @@ export interface IEmailField extends IField {
   isVerifiable: boolean
 }
 
-export interface IEmailFieldSchema extends IEmailField, IFieldSchema {}
+export interface IEmailFieldSchema extends IEmailField, IFieldSchema {
+  isVerifiable: boolean
+}
```
Comment (mantariksh) [inline src/types/field/emailField.ts:18]: @karrui why was this necessary? isn't this already in `IEmailField`

Comment (karrui) [reply src/types/field/emailField.ts:18]: Sigh, pain point with discriminators. `IFieldSchema` once again overrode `isVerifiable` back to an optional field, but we want it to be correctly required, so we need to override it again.
